### PR TITLE
pix2struct for doc vqa notebook

### PR DESCRIPTION
**Pull request by eaidova**
ticket: CVS-123223

### CODE REVIEW

File: notebooks/260-pix2struct-docvqa/260-pix2struct-docvqa.ipynb [hunk 1]
```diff
@@ -0,0 +1,381 @@
...
+    "import gc\n",
+    "from pathlib import Path\n",
+    "from optimum.intel.openvino import OVModelForPix2Struct\n",
+    "\n",
+    "model_id = \"google/pix2struct-docvqa-base\"\n",
+    "model_dir = Path(model_id.split('/')[-1])\n",
+    "\n",
+    "if not model_dir.exists():\n",
+    "    ov_model = OVModelForPix2Struct.from_pretrained(model_id, export=True, compile=False)\n",
+    "    ov_model.half()\n",
```
Comment (aleksandr-mokrov) [inline notebooks/260-pix2struct-docvqa/260-pix2struct-docvqa.ipynb:122]: Add a description what does `half()` do

File: notebooks/260-pix2struct-docvqa/260-pix2struct-docvqa.ipynb [hunk 2]
```diff
@@ -0,0 +1,381 @@
...
+    "processor = Pix2StructProcessor.from_pretrained(model_id)\n",
+    "ov_model = OVModelForPix2Struct.from_pretrained(model_dir, device=device.value)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "83a2ac29-80d9-4df0-b4e5-584a735f09e7",
+   "metadata": {},
+   "source": [
+    "Let's see model in action. For testing model, we will use screenshot from [OpenVINO documentation](https://docs.openvino.ai/2023.1/get_started.html#openvino-advanced-features)"
```
Comment (aleksandr-mokrov) [inline notebooks/260-pix2struct-docvqa/260-pix2struct-docvqa.ipynb:217]: Looks like it should be `the model`. And in other cases too.

File: notebooks/260-pix2struct-docvqa/260-pix2struct-docvqa.ipynb [hunk 3]
```diff
@@ -0,0 +1,381 @@
...
...qWTmZq12LtB1yO6rdSbUrIjT3oNbsK5d2n2k5v7sDlv7y5IZTsRREMphg9v2m9L05/eDUSUUjB/jba3HyPr97muU2a2JrHVUWVzwyYHZde3keWfuqQL/r5DZVy7eYzYeN8rq3cd30VQUj2znQM1+c2n02w6b/gmDRUl+V6slR+CYHqH7Q6EGnfjv4+3TmuKH+loLPD47uvlHm9tuoVsI5BA2fIUMb3927dkujlRPaWAkSbm/ddKuAihiKq6Z7jpzW6fGipeMqowZ1bmqtxS/KSY97+zTJaerSvtJ3RRRpqXzAV5SbTKM4t/DWvXDz6KVOC3s4UlKfHF55NJJ4xExuVFsHa+Txg4v33DuYMylaFk7WunU3GSkM1Q4c+1vzl2tX/IZ8Ht7Zw4RSnB71Olyjx4KBruLwCDPu0Cdg9/xbkUz/xZ2U/dkKlN3Exwk9ePbQc69xzdkln/7ZtPZOMVPhTDtZcdUvMd6O9r/8xxbiCzUzg53o6U/3rTqfTJU+S6nK+AV7Kocvxc6svBbV/qysnFoAaxZWPTaqdGakpodoCytQOuzXrL1uKQrGXpmFYhJZFDuvJuz9Z18VWg8XT0xqe3o75m4K4Wm2/a3aZ9XqVrsw9xf30Nqrork1b8JYe+nw3cAZQebchDs7l11Il1n8UbuAqhw17GKluqdXlfyWeyi77Yh+x8efXLiINbGfN/EdlesHTsWxiHVeVWHMt6yuTrJqhLp1Ki2Xme487M+9hgf2n729949jxQK6vkWD5mM3jRvaUvwIBtVxdDN/fOvMtr8y88pwlrGDV68Vv//qJ3ybCNX2nrpnr/2BA38dX3Uuq4KibWBq6+bbUdnbkKi+z5TdRz1PHzpz6/TmS/mVFG1TJ68+S5YND3aVWQdELPFqQnwi8d2W7fN2ZnC0bFuM3jDrF/LNJobryK0HLQ/vO3V+07X0Ypylb2zdoHmbvsSNgdob8Vd/WhvcvJhr1q6zZ82FzsrFoHqtfxnoen+XzGQPMVKYtl100KrZ8SMX7h1Zd7aIS/whKVefEWtGDmljJ70nlROJGrq66YU+PLbpZFZBOaZt7tJ85PqJwxvWdYSVkyk6wIw7Lt1D27/t0F+rZ+6uZBo7+QxfP3l0oPgVNOIOZM72ZTs27J3Vc2eFhknD1kPm/CKYs5uUUkc7ksWIX8yo3e/7tRx3Hby8a/HxQj7TwM6
```
Comment (aleksandr-mokrov) [inline notebooks/260-pix2struct-docvqa/260-pix2struct-docvqa.ipynb:245]: Looks complicated for this tutorial. It is always URL starting with `https`

File: notebooks/260-pix2struct-docvqa/260-pix2struct-docvqa.ipynb [hunk 4]
```diff
@@ -0,0 +1,381 @@
...
+    "\n",
+    "[Pix2Struct](https://arxiv.org/pdf/2210.03347.pdf) is a multimodal model for understanding visually-situated language that easily copes with extracting information from images. The model is trained using the novel learning technique to parse masked screenshots of web pages into simplified HTML, providing a significantly well-suited pretraining data source for the range of downstream activities such as OCR, visual question answering and image captioning.\n",
+    "\n",
+    "In this tutorial we consider how to run Pix2Struct model using OpenVINO for solving document visual question answering task. We will use pre-trained model from the [Hugging Face Transformers](https://huggingface.co/docs/transformers/index) library. To simplify the user experience, the [Hugging Face Optimum](https://huggingface.co/docs/optimum) library is used to convert the model to OpenVINO™ IR format.\n",
+    "\n",
+    "The tutorial consist of the following steps:\n",
```
Comment (aleksandr-mokrov) [inline notebooks/260-pix2struct-docvqa/260-pix2struct-docvqa.ipynb:21]: It probably makes sense to combine these steps with `Table of content`

File: notebooks/260-pix2struct-docvqa/260-pix2struct-docvqa.ipynb (hunk 3 above)
[the remark this answers is not in the document]
Comment (eaidova) [reply notebooks/260-pix2struct-docvqa/260-pix2struct-docvqa.ipynb:314]: it is valid answer, please pay attention on bold text in braces for this time.
But yes, maybe better to select another question

File: notebooks/260-pix2struct-docvqa/260-pix2struct-docvqa.ipynb [hunk 5]
```diff
@@ -0,0 +1,381 @@
...
+      "Compiling the decoder to AUTO ...\n",
+      "Compiling the decoder to AUTO ...\n"
+     ]
+    }
+   ],
+   "source": [
+    "from transformers import Pix2StructProcessor\n",
+    "\n",
+    "processor = Pix2StructProcessor.from_pretrained(model_id)\n",
+    "ov_model = OVModelForPix2Struct.from_pretrained(model_dir, device=device.value)"
```
[the remark this answers is not in the document]
Comment (eaidova) [reply notebooks/260-pix2struct-docvqa/260-pix2struct-docvqa.ipynb:209]: I'm asking GPU team to check that this check still actual on optimum side
I do not understand your comment about installing torch without cuda. How it is connected? For running model via openvino requires to obtain model in IR first, there is no any difference which device you use for torch model tracing before running openvino, from operations perspective model representations are the same, changed only location of tensors and internal details how operation implemented (general interface and signature are the same -  it is enough for conversion). After model converted, openvino do not uses torch and have no dependency on cuda, moreover GPU device not always means cuda at all, it maybe intel GPU that does not supported by torch.

So installing torch happens in notebook only if user do not have torch installed yet, so why we need to install full version that in 4 times larger and installs additionally some nvidia packages if we do not plan to use it in notebook?

File: notebooks/260-pix2struct-docvqa/260-pix2struct-docvqa.ipynb (hunk 3 above)
[the remark this answers is not in the document]
Comment (aleksandr-mokrov) [reply notebooks/260-pix2struct-docvqa/260-pix2struct-docvqa.ipynb:314]: In my understanding the exhibition opens at 11:14. But maybe "11:14 to 11:39 a.m." makes sense too.
Nevertheless:
When does exsibition open? (default question) - 11:14 to 11:39 a.m.
When does exhibition open? - 12:58 to 4:00 p.m.
When does the exhibition open? - 11:39 a.m.
When is the exhibition open? - 12:58 to 4:00 p.m.
When does exhibits open? - 12:58 to 4:00 p.m.

[the remark this answers is not in the document]
Comment (eaidova) [reply notebooks/260-pix2struct-docvqa/260-pix2struct-docvqa.ipynb:338]: I changed example question for avoiding confusion

File: notebooks/260-pix2struct-docvqa/260-pix2struct-docvqa.ipynb (hunk 1 above)
Comment (eaidova) [reply notebooks/260-pix2struct-docvqa/260-pix2struct-docvqa.ipynb:116]: added

File: notebooks/260-pix2struct-docvqa/260-pix2struct-docvqa.ipynb (hunk 2 above)
Comment (eaidova) [reply notebooks/260-pix2struct-docvqa/260-pix2struct-docvqa.ipynb]: articles added and verified with grammarly

File: notebooks/260-pix2struct-docvqa/260-pix2struct-docvqa.ipynb (hunk 4 above)
Comment (eaidova) [reply notebooks/260-pix2struct-docvqa/260-pix2struct-docvqa.ipynb]: it looks reductant, removed and stayed only table of content

File: notebooks/260-pix2struct-docvqa/260-pix2struct-docvqa.ipynb (hunk 3 above)
Comment (eaidova) [reply notebooks/260-pix2struct-docvqa/260-pix2struct-docvqa.ipynb]: simpilfied

File: notebooks/260-pix2struct-docvqa/260-pix2struct-docvqa.ipynb (hunk 5 above)
[the remark this answers is not in the document]
Comment (eaidova) [reply notebooks/260-pix2struct-docvqa/260-pix2struct-docvqa.ipynb:203]: for now, I stayed device selection (for simplification adding, when optimum will be fixed), but exclude gpu

File: notebooks/260-pix2struct-docvqa/260-pix2struct-docvqa.ipynb [hunk 6]
```diff
@@ -0,0 +1,381 @@
...
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "id": "1414901f-1b26-442d-90f2-55b2a28ddf7b",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%pip install torch torchvision torchaudio --index-url https://download.pytorch.org/whl/cpu\n",
```
[the remark this answers is not in the document]
Comment (eaidova) [reply notebooks/260-pix2struct-docvqa/260-pix2struct-docvqa.ipynb]: added

File: notebooks/260-pix2struct-docvqa/260-pix2struct-docvqa.ipynb [hunk 7]
```diff
@@ -0,0 +1,401 @@
...
+    "\n",
+    "[Pix2Struct](https://arxiv.org/pdf/2210.03347.pdf) is a multimodal model for understanding visually situated language that easily copes with extracting information from images. The model is trained using the novel learning technique to parse masked screenshots of web pages into simplified HTML, providing a significantly well-suited pretraining data source for the range of downstream activities such as OCR, visual question answering, and image captioning.\n",
+    "\n",
+    "In this tutorial, we consider how to run the Pix2Struct model using OpenVINO for solving document visual question answering task. We will use a pre-trained model from the [Hugging Face Transformers](https://huggingface.co/docs/transformers/index) library. To simplify the user experience, the [Hugging Face Optimum](https://huggingface.co/docs/optimum) library is used to convert the model to OpenVINO™ IR format.\n",
+    "\n",
+    "### Table of content:\n",
+    "- [About Pix2Struct](#About-Pix2Struct-Uparrow)\n",
```
Comment (aleksandr-mokrov) [inline notebooks/260-pix2struct-docvqa/260-pix2struct-docvqa.ipynb:22]: Seems it doesn't work in all application. In jupyterlab it doesn't switched to About Pix2Struct header. But` - [About Pix2Struct](#About-Pix2Struct-$\Uparrow$)` works.

File: notebooks/260-pix2struct-docvqa/260-pix2struct-docvqa.ipynb [hunk 8]
```diff
@@ -0,0 +1,401 @@
...
+    "%pip install -q torch torchvision torchaudio --index-url https://download.pytorch.org/whl/cpu\n",
+    "%pip install -q \"git+https://github.com/huggingface/optimum-intel.git\" \"openvino>=2023.1.0\" transformers onnx gradio"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "2dfce1e7-7d67-4e5a-aec5-74485430fe78",
+   "metadata": {},
+   "source": [
+    "## Download and Convert Model [$\\Uparrow$](#Table-of-contents:) [$\\Uparrow$](#Table-of-content:)\n",
```
Comment (aleksandr-mokrov) [inline notebooks/260-pix2struct-docvqa/260-pix2struct-docvqa.ipynb:69]: First `[$\\Uparrow$](#Table-of-contents:)` is redundant.

**Review by aleksandr-mokrov — APPROVED** [review]

File: notebooks/260-pix2struct-docvqa/260-pix2struct-docvqa.ipynb [hunk 9]
```diff
@@ -0,0 +1,370 @@
...
+    "\n",
+    "[Pix2Struct](https://arxiv.org/pdf/2210.03347.pdf) is a multimodal model for understanding visually situated language that easily copes with extracting information from images. The model is trained using the novel learning technique to parse masked screenshots of web pages into simplified HTML, providing a significantly well-suited pretraining data source for the range of downstream activities such as OCR, visual question answering, and image captioning.\n",
+    "\n",
+    "In this tutorial, we consider how to run the Pix2Struct model using OpenVINO for solving document visual question answering task. We will use a pre-trained model from the [Hugging Face Transformers](https://huggingface.co/docs/transformers/index) library. To simplify the user experience, the [Hugging Face Optimum](https://huggingface.co/docs/optimum) library is used to convert the model to OpenVINO™ IR format.\n",
+    "\n",
+    "### Table of content:\n",
+    "- [About Pix2Struct](#About-Pix2Struct-$\\Uparrow$)\n",
+    "- [Prerequisites](#Prerequisites-$\\Uparrow$)\n",
+    "- [Download and Convert Model](#Download-and-Convert-Model-Uparrow)\n",
```
Comment (aleksandr-mokrov) [inline notebooks/260-pix2struct-docvqa/260-pix2struct-docvqa.ipynb:24]: Not all `Uparrow` have been replaced

**Pull request merged**